python/paddle/tensor/math.py

```diff
@@ -2613,6 +2613,52 @@ def atan2(x, y, name=None):
                 type='atan2', inputs=inputs, outputs={'Out': out})
         return out
 
+def lerp(x, y, weight, name=None):
+    r"""
+    Does a linear interpolation between x and y based on weight.
+
+    Equation:
+        .. math::
+
+            lerp(x, y, weight) = x + weight * (y - x).
+
+    Args:
+        x (Tensor): An N-D Tensor, the data type is float32, float64.
+        y (Tensor): An N-D Tensor, the data type is float32, float64.
+        weight (float|Tensor): the weight for the interpolation formula.
+        name (str, optional): Name for the operation (optional, default is None). For more information, please refer to :ref:`api_guide_Name`.
+
+    Returns:
+        out (Tensor): An N-D Tensor, the shape and data type is the same with input.
+
+    Example:
+        .. code-block:: python
+
+            import paddle
+            
+            x = paddle.arange(1., 5., dtype='float32')
+            y = paddle.empty([4], dtype='float32')
+            y.fill_(10.)
+            out = paddle.lerp(start, end, 0.5)
+            # out: [5.5., 6., 6.5, 7.]
+
+    """
+    if in_dygraph_mode():
+        check_type(weight, 'weight', (float, paddle.Tensor, Variable), 'lerp')
+        if isinstance(weight, float):
+            weight = paddle.to_tensor(weight, dtype=x.dtype, place=paddle.CPUPlace())
```

Comment:
If we are not limited to `CPUPlace`, will we get some speed up for GPU?

Reply:
done